src/components/app/ErrorBoundary.js

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import React from 'react';
+import { reportError } from '../../utils/analytics';
+import './ErrorBoundary.css';
+
+type State = {|
+  hasError: boolean,
+  showDetails: boolean,
+  error?: Error,
+  componentStack?: string,
+|};
+
+type Props = {|
+  children: *,
+  message: string,
+|};
+
+/**
+ * This component will catch errors in components, and display a more friendly error
+ * message. This stops the entire app from unmounting and displaying an empty screen.
+ *
+ * See: https://reactjs.org/docs/error-boundaries.html
+ */
+export class ErrorBoundary extends React.Component<Props, State> {
+  state = {
+    hasError: false,
+    showDetails: false,
+  };
+
+  componentDidCatch(
+    error: any,
```

Comment:
I'd rather use `mixed` than `any`. Because of `any` we're lying in the state which is a `Error` because Flow didn't catch the mismatch.

Although Flow seems to think it's an Error: https://github.com/facebook/flow/blob/master/lib/react.js#L78
So we could as well do the same.

(I don't know if our Flow contains this definition)

Reply:
Mixed works for me. Technically you can throw anything, so it may not be an `Error`.